lib/Sema/DerivedConformances.h

```diff
@@ -18,19 +18,24 @@
 #ifndef SWIFT_SEMA_DERIVEDCONFORMANCES_H
 #define SWIFT_SEMA_DERIVEDCONFORMANCES_H
 
+#include "swift/AST/Identifier.h"
```

Comment:
Headers should not include other headers if possible. Please put this include in DerivedConformanceCodable.cpp.

Reply:
In this case b/c Identifier is passed by value, it can not be forward declared.

Reply:
I'm not sure what you mean. The header file does not reference anything from Identifier.h, only DerivedConformanceCodable.cpp does. So DerivedConformanceCodable.cpp should include Identifier.h. You don't need to forward-declare anything in DerivedConformances.h.

Reply:
/// Declare a read-only property.                                                                                                                                                                                                     
std::pair<VarDecl *, PatternBindingDecl *>
declareDerivedProperty(TypeChecker &tc,
                       Decl *parentDecl,
                       NominalTypeDecl *typeDecl,
                       Identifier name,
                       Type propertyInterfaceType,
                       Type propertyContextType,
                       bool isStatic,
                       bool isFinal);

Reply:
Oh ok, so the reference was not introduced by your changes, you're just cleaning it up. Got it. A separate commit titled "header gardening" or similar would've made that clearer.

Reply:
I agree. TBH this was just clang formatting fall out that I quickly worked around.